src/DistributedLock.Postgres/PostgresAdvisoryLock.cs

```diff
@@ -207,7 +211,13 @@ private static async ValueTask<bool> HasTransactionAsync(DatabaseConnection conn
 
     private async ValueTask ReleaseAsync(DatabaseConnection connection, PostgresAdvisoryLockKey key, bool isTry)
     {
-        Invariant.Require(!UseTransactionScopedLock(connection));
+        // For transaction scoped advisory locks, the lock can only be released by ending the transaction.
+        // If the transaction is internally-owned, then the lock will be released when the transaction is disposed as part of the internal connection management.
+        // If the transaction is externally-owned, then the lock will have to be released explicitly by the transaction initiator.
```

Comment:
I'm not comfortable with these semantics; it's just too different from how the other locks work to say that releasing the handle does not release the lock. This feels like the kind of thing that will be hard to discover, since correct-looking code will just be wrong and I don't like having to add `except with pg externally-owned-transaction-scoped-locks!` to all the generic code examples.

The static utility method feels like a better model for what we're trying to do here, which is to apply a one-way change to a transaction without any notion of a returned disposable scope.

Reply:
Ok, I can see why a static method in the API may be a better option in this case. I currently have some worries regarding how exactly it will be implemented, but I'll try.

Reply:
Hi @madelson, I've finally managed to look into the static utility methods. I still didn't add summay comments for the static methods, and I need to revert the change in the ReleaseAsync method in the PostgresAdvisoryLock class, but please take a look at the recent changes and tell me if I am on the right track.

Reply:
@Tzachi009 apologies for the long delay. The new static methods look like they're on the right track. I left a few comments.

Reply:
Hi @madelson, thanks for the feedback.

- I replied and fixed to your comments (with the exception of one, where what you asked is not possible).
- I reverted my changes in the `ReleaseAsync()` method in the `PostgresAdvisoryLock` class.
- Added summaries for the static utility methods.
- Added some unit tests - I hope it's enough, considering I didn't added a lot of code here.